### PR TITLE
New package: cortexcommand-6.2.2

### PR DESCRIPTION
**Pull request by xolophreny**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**, compiled

#### Local build testing
- I built this PR locally for my native architecture, x86_64
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64 (cross)
  - i686 (fails, meson can't find tbb, also doesn't seem to be supported upstream)
  - x86_64-musl (fails, fmod game audio library needs glibc)

Needs #50375 for recent enough LuaJIT, otherwise builds but crashes in use

### CODE REVIEW

**Comment by Emru1** [comment]
> Needs https://github.com/void-linux/void-packages/pull/50375 for recent enough LuaJIT, otherwise builds but crashes in use
Oh, that was the reason why it was constantly crashing

**Comment by Calandracas606** [comment]
>  686 (fails, meson can't find tbb, also doesn't seem to be supported upstream)

maybe because void's tbb package only builds the python bindings on `x86_64*|aarch64*`


>  x86_64-musl (fails, fmod needs glibc)

musl's `math.h` should contain fmod: https://git.musl-libc.org/cgit/musl/tree/include/math.h#n227
you might need to patch the software to `#include <math.h>` where fmod is needed

**Comment by Bnyro** [comment]
> musl's math.h should contain fmod
>
It's about the `fmod` game audio library (https://www.fmod.com/), not the mathematical function.

**Comment by xolophreny** [comment]
> maybe because void's tbb package only builds the python bindings on `x86_64*|aarch64*`

Tried adding `i686*` to that list in tbb template and building tbb-devel, didn't help, still `Run-time dependency tbb found: NO (tried pkgconfig)`, and unfortunately no other ideas

> musl's `math.h` should contain fmod

Yeah sorry wasn't clear enough there, this is indeed about the fmod game audio library which this uses

**Comment by Calandracas606** [comment]
> It's about the `fmod` game audio library (https://www.fmod.com/), not the mathematical function.

This looks like a proprietary library.

If a binary is bundled in the distfiles, that would make this package nonfree.

additionally, this package being AGPL but using a proprietary binary would make distribution legally dubious.

maybe the package would need to be restricted

**Comment by Calandracas606** [comment]
> Tried adding `i686*` to that list in tbb template and building tbb-devel, didn't help, still `Run-time dependency tbb found: NO (tried pkgconfig)`, and unfortunately no other ideas
> 

Try adding cmake to hostmakedepends, and/or patching meson.build to use something like this:

```
tbb_dep = dependency(
  'TBB',
  method : 'cmake',
  modules : [
    'TBB::tbb',
    'TBB::tbbmalloc',
    'TBB::tbbmalloc_proxy',
  ],
  required : true,
)
```

**Comment by xolophreny** [comment]
> Try adding cmake to hostmakedepends, and/or patching meson.build to use something like this:

Still results in `Run-time dependency tbb found: NO (tried cmake)`

Fortunately, looks like they only provide fmod blobs for x86_64 and arm64, so I'm assuming that it would fail from that anyway

As for the blobs - asked the devs about it, first reply was that it should be alright to distribute these final useful bits of fmod, and [this fmod forum thread](https://qa.fmod.com/t/how-to-handle-fmod-in-an-open-source-project/21208) seems to suggest so as well as long as it's not the whole library, but uh, I'll agree with that it sounds dubious and package should probably be restricted

**Comment by Calandracas606** [comment]
> As for the blobs - asked the devs about it, first reply was that it should be alright to distribute these final useful bits of fmod, and [this fmod forum thread](https://qa.fmod.com/t/how-to-handle-fmod-in-an-open-source-project/21208) seems to suggest so as well as long as it's not the whole library, but uh, I'll agree with that it sounds dubious and package should probably be restricted

if this is the case, void would not be able to distribute the software under the terms of the AGPL

File: srcpkgs/cortexcommand/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+configure_args="$(vopt_if debug --buildtype=debug)"
+hostmakedepends="pkg-config"
+makedepends="SDL2-devel SDL2_image-devel tbb-devel libflac-devel LuaJIT-devel minizip-devel liblz4-devel libpng-devel"
+depends="LuaJIT>=2.1.1"
+short_desc="Cortex Command Community Project"
+maintainer="Nika Jay <xolophreny@proton.me>"
+license="AGPL-3.0-only, custom:Proprietary"
+homepage="https://cortex-command-community.github.io"
+distfiles="https://github.com/cortex-command-community/Cortex-Command-Community-Project/archive/refs/tags/v${version}.tar.gz"
+checksum=66fb83566982d2fed4b01c2c83b5cd2ef4b8e6ede08f160eb16509f4bf340c80
```
Comment (Bnyro) [inline srcpkgs/cortexcommand/template:16]:
```suggestion
checksum=66fb83566982d2fed4b01c2c83b5cd2ef4b8e6ede08f160eb16509f4bf340c80
repository=nonfree
```

**Review by Bnyro — APPROVED** [review]

**Pull request closed without merging**